### PR TITLE
fix some vm issues found during testing

### PR DESCRIPTION
**Pull request by lazedo**
type checking for older messages
maybe_load_vmboxes recursion

### CODE REVIEW

File: core/kazoo_voicemail/src/kz_vm_message.erl [hunk 1]
```diff
@@ -43,6 +43,7 @@
 -define(MODB_COUNT_VIEW, <<"mailbox_messages/count_per_folder">>).
 -define(COUNT_BY_VMBOX, <<"mailbox_messages/count_by_vmbox">>).
 -define(BOX_MESSAGES_CB_LIST, <<"vmboxes/crossbar_listing">>).
+-define(PVT_LEGACY_TYPE, <<"private_media">>).
```
Comment (fenollp) [inline core/kazoo_voicemail/src/kz_vm_message.erl]: Is string missing "pvt_" ?

Comment (lazedo) [inline core/kazoo_voicemail/src/kz_vm_message.erl]: what ?

Comment (fenollp) [inline core/kazoo_voicemail/src/kz_vm_message.erl]: macro is called `PVT_LEGACY_TYPE` but content reads `private_media`. I was wondering then why call it PVT_ ?

Comment (lazedo) [inline core/kazoo_voicemail/src/kz_vm_message.erl]: really ?
[check here](https://github.com/2600hz/kazoo/blob/master/core/kazoo_documents/src/kzd_callflow.erl#L22)
[or here](https://github.com/2600hz/kazoo/blob/master/core/kazoo_documents/src/kz_notification.erl#L48)
or any other in kazoo_documents

**Comment by lazedo** [comment]
@fenollp pull request was assigned to @icehess

**Comment by fenollp** [comment]
ah sorry I missed that. Want be to restore?

**Comment by lazedo** [comment]
@fenollp no worries, @icehess can look at the commit and correct anything that he may find.

**Pull request merged**